### PR TITLE
Cross target NETSTANDARD2.0

### LINKED ISSUE #78: Cross target NETSTANDARD2.0
The official .NET team recommends cross targeting .NETSTANDARD2.0 for all libraries. This reduces the dependency tree and possibility of a library conflict. 

REFERENCE:

https://github.com/dotnet/standard/blob/master/docs/versions.md#how-do-i-know-which-net-standard-version-i-should-target

- You should target the lowest version you can get away with.
- If you target .NET Standard 1.x, add .NET Standard 2.0 as another target.

### PR DESCRIPTION
**Pull request by gitvinit**
This update will help cross target NETSTANDARD2.0

# Summary

# Checklist

- [x] Added changelog entry
- [x ] Ran unit tests (See README for instructions)

fixes #78

### CODE REVIEW

File: src/Braintree/Braintree.csproj [hunk 1]
```diff
@@ -45,10 +44,19 @@
...
 
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+    <DefineConstants>$(DefineConstants);netcore</DefineConstants>
+  </PropertyGroup>
+
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
   </ItemGroup>
 
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
```
Comment (jackellenberger) [inline src/Braintree/Braintree.csproj]: Would you mind just combining this with the netstandard1.3 ItemGroup if there's no difference in the versions?

Comment (gitvinit) [reply src/Braintree/Braintree.csproj]: @jackellenberger combined the itemgroups with new commit

**Comment by iamcarbon** [comment]
Can we also update Microsoft.AspNetCore.Http.Abstractions to 2.2?

File: src/Braintree/Braintree.csproj [hunk 2]
```diff
@@ -45,10 +44,19 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="1.0.2" />
```
Comment (iamcarbon) [inline src/Braintree/Braintree.csproj]: Update to 2.2.0?

File: src/Braintree/Braintree.csproj [hunk 3]
```diff
@@ -45,10 +44,19 @@
...
+    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.0.1" />
+  </ItemGroup>
+
   <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="1.0.2" />
```
Comment (iamcarbon) [inline src/Braintree/Braintree.csproj]: Update to 2.2.0?

File: src/Braintree/Braintree.csproj [hunk 4]
```diff
@@ -45,10 +44,19 @@
...
     <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.0.1" />
```
Comment (iamcarbon) [inline src/Braintree/Braintree.csproj]: Update to 4.3.0?

**Comment by gitvinit** [comment]
@iamcarbon thanks for reviewing the change and for the package update suggestion Microsoft.AspNetCore.Http.Abstractions 2.2.0 is not compatible with netstandard1.3 should I separate the package references by netstandard version?
`<ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.3.0" />
  </ItemGroup>`
and netstandard1.3 will have
`<ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="1.0.2" />
    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.0.1" />
  </ItemGroup>`

**Comment by iamcarbon** [comment]
Curious if we can just remove the Microsoft.AspNetCore.Http.Abstractions dependency all together. I just did this locally and all tests continue to pass. It doesn't look like anything is using it.

@jackellenberger -- do you have ideas on why this dependency was introduced or know of any unintended consequences if we remove it? If we keep it, it's going to introduce the possibility for binding redirect problems.

**Comment by jackellenberger** [comment]
@iamcarbon The Abstractions dependency was added back when we started supporting .net core 1.x. I'm not sure if anything has changed regarding it being a requirement, but we can take it out and see if anything complains. I don't see anything asking for it explicitly.

File: src/Braintree/Braintree.csproj (hunk 4 above)
Comment (iamcarbon) [reply src/Braintree/Braintree.csproj]: Update: Remove.

File: src/Braintree/Braintree.csproj (hunk 3 above)
Comment (iamcarbon) [reply src/Braintree/Braintree.csproj]: Update: Remove.

File: src/Braintree/Braintree.csproj (hunk 4 above)
Comment (gitvinit) [reply src/Braintree/Braintree.csproj]: Removed Microsoft.AspNetCore.Http.Abstractions updated System.Xml.XPath.XmlDocument to 4.3.0

File: src/Braintree/Braintree.csproj [hunk 5]
```diff
@@ -17,8 +17,7 @@
       * Add `Username` and `VenmoUserId` to `PaymentMethodNonceDetails`.
     </PackageReleaseNotes>
     <PackageProjectUrl>https://github.com/braintree/braintree_dotnet</PackageProjectUrl>
-    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netstandard1.3' ">$(PackageTargetFallback);dnxcore50;portable-net45+win7</PackageTargetFallback>
```
Comment (jackellenberger) [inline src/Braintree/Braintree.csproj]: @iamcarbon @gitvinit sorry I'm dragging my feet on this one, I just want to make sure we're not losing core 1.X or `net451` support here. It seems like this line _needs_ to be removed for netstandard packages, but that kicks the minimum framework version up to `net461` (I think). Thanks for your patience.

**Comment by gitvinit** [comment]
@jackellenberger thanks for approving the changes updated changelog and reverted changes for .sln file @iamcarbon thanks for your help throughout the process

**Pull request merged**